audio_service/test/mock_base_audio_handler.dart

```diff
@@ -0,0 +1,421 @@
+  set stubMediaItem(BehaviorSubject<MediaItem?> value) {
+    _stubMediaItem.value = value;
+  }
+
+  final _Stub<BehaviorSubject<RatingStyle>> _stubRatingStyle = _Stub();
+  BehaviorSubject<RatingStyle> get stubRatingStyle =>
+      _stubRatingStyle.exposeValue;
+  set stubRatingStyle(BehaviorSubject<RatingStyle> value) {
+    _stubRatingStyle.value = value;
+  }
+
+  final _Stub<BehaviorSubject<AndroidPlaybackInfo>> _stubAndroidPlaybackInfo =
+      _Stub();
+  BehaviorSubject<AndroidPlaybackInfo> get stubAndroidPlaybackInfo =>
+      _stubAndroidPlaybackInfo.exposeValue;
+  set stubAndroidPlaybackInfo(BehaviorSubject<AndroidPlaybackInfo> value) {
+    _stubAndroidPlaybackInfo.value = value;
+  }
+
+  final _Stub<PublishSubject<Object?>> _stubCustomEvent = _Stub();
+  PublishSubject<Object?> get stubCustomEvent => _stubCustomEvent.exposeValue;
+  set stubCustomEvent(PublishSubject<Object?> value) {
+    _stubCustomEvent.value = value;
+  }
+
+  final _Stub<BehaviorSubject<Object?>> _stubCustomState = _Stub();
+  BehaviorSubject<Object?> get stubCustomState => _stubCustomState.exposeValue;
+  set stubCustomState(BehaviorSubject<Object?> value) {
+    _stubCustomState.value = value;
+  }
+
+  final _Stub<Object?> _stubCustomAction = _Stub();
+  Object? get stubCustomAction => _stubCustomAction.exposeValue;
+  set stubCustomAction(Object? value) {
+    _stubCustomAction.value = value;
+  }
+
+  final _Stub<List<MediaItem>> _stubGetChildren = _Stub();
+  List<MediaItem> get stubGetChildren => _stubGetChildren.exposeValue;
+  set stubGetChildren(List<MediaItem> value) {
+    _stubGetChildren.value = value;
+  }
+
+  final _Stub<BehaviorSubject<Map<String, dynamic>>> _stubSubscribeToChildren =
+      _Stub();
+  BehaviorSubject<Map<String, dynamic>> get stubSubscribeToChildren =>
+      _stubSubscribeToChildren.exposeValue;
+  set stubSubscribeToChildren(BehaviorSubject<Map<String, dynamic>> value) {
+    _stubSubscribeToChildren.value = value;
+  }
+
+  final _Stub<MediaItem?> _stubGetMediaItem = _Stub();
+  MediaItem? get stubGetMediaItem => _stubGetMediaItem.exposeValue;
+  set stubGetMediaItem(MediaItem? value) {
+    _stubGetMediaItem.value = value;
+  }
+
+  final _Stub<List<MediaItem>> _stubSearch = _Stub();
+  List<MediaItem> get stubSearch => _stubSearch.exposeValue;
+  set stubSearch(List<MediaItem> value) {
+    _stubSearch.value = value;
+  }
+
+  void reset() {
+    _stubPlaybackState.reset();
+    _stubQueue.reset();
+    _stubQueueTitle.reset();
+    _stubMediaItem.reset();
+    _stubRatingStyle.reset();
+    _stubAndroidPlaybackInfo.reset();
+    _stubCustomEvent.reset();
+    _stubCustomState.reset();
+    _stubCustomAction.reset();
+    _stubGetChildren.reset();
+    _stubSubscribeToChildren.reset();
+    _stubGetMediaItem.reset();
+    _stubSearch.reset();
+    log.clear();
+    argumentsLog.clear();
+  }
+
+  void _log(String method, [List<Object?> arguments = const [null]]) {
```

Comment:
This actually should be an empty array instead. I have already changed this on another branch that adds platform connection tests over this PR, so I keep it as is and will change it in my next PR